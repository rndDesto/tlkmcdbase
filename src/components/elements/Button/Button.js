import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Button(props) {
  const { buttonProps, children, className, onClick } = props;
  const classes = [styles.root, className].filter(Boolean).join(' ');

  return (
    <button className={classes} onClick={onClick} {...buttonProps}>
      {children}
    </button>
  );
}

Button.defaultProps = {
  buttonProps: {},
  children: null,
  className: '',
  onClick: () => {},
};

Button.propTypes = {
  buttonProps: PropTypes.object,
  children: PropTypes.node,
  className: PropTypes.string,
  onClick: PropTypes.func,
};
