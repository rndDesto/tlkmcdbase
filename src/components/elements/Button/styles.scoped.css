.root {
  background-color: var(--primary-color);
  border: none;
  border-radius: 0.25rem;
  color: #FFFFFF;
  cursor: pointer;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.125rem;
  padding: 0.9375rem;
}
