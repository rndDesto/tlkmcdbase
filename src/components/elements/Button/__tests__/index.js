import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Button from '../Button';

describe('src/components/elements/Button', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    Button.defaultProps.onClick();
    const tree = shallow.render(<Button />);
    expect(tree).toMatchSnapshot();
  });
});
