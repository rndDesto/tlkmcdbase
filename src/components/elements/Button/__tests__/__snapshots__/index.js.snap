// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Button render 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;
